src/librustc/util/ppaux.rs

```diff
@@ -907,6 +907,27 @@ impl<'tcx> fmt::Display for ty::TypeVariants<'tcx> {
             }
             TyTrait(ref data) => write!(f, "{}", data),
             ty::TyProjection(ref data) => write!(f, "{}", data),
+            ty::TyAnon(def_id, substs) => {
+                ty::tls::with(|tcx| {
+                    // Grab the "TraitA + TraitB" from `impl TraitA + TraitB`,
```

Comment:
As an aside, I suspect that the grammar ought to be `impl (TraitA + TraitB)` -- at least, the precedence of `impl TraitA + TraitB` looks surprising to me. Though maybe I could learn to like it. 

I'm imagining things like `F: Fn() -> impl Trait + Send` <-- how should this be parsed?